container/rules/docker_toolchains.bzl

```diff
@@ -62,16 +63,43 @@ def _language_tool_layer_impl(ctx, symlinks=None, env=None, tars=None,
   files = files or ctx.files.files
   packages = packages or ctx.attr.packages
   additional_repos = additional_repos or ctx.attr.additional_repos
-  installables_tars = installables_tars or []
-  if installables_tars == [] and ctx.file.installables_tar:
-    installables_tars = [ctx.file.installables_tar]
   installation_cleanup_commands = installation_cleanup_commands or ctx.attr.installation_cleanup_commands
+  installables_tar = None
+
+  # If a link to the tarball of deb packages is passed in through
+  # --action_env=DOWNLOADABLE_TAR=<link>, use that one.
+  if "DOWNLOADABLE_TAR" in ctx.configuration.default_shell_env:
+    if "DOWNLOADABLE_TAR_SHA256" not in ctx.configuration.default_shell_env:
```

Comment:
this should be in the input validation

Reply:
action_envs seems can't be accessed in macro level (i.e. from kwargs).